packages/common/src/intl/intlHelpers.ts

```diff
@@ -0,0 +1,48 @@
+import { useIntl } from 'react-intl';
+import type { PrimitiveType } from 'intl-messageformat';
+
+// "import type" ensures en messages aren't bundled by default
+import * as sourceOfTruth from './locales/en.json';
+
+export type SupportedLocales = 'en' | 'fr' | 'pt';
+export type LocaleMessages = typeof sourceOfTruth;
+export type LocaleKey = keyof LocaleMessages;
+
+export const useFormatMessage = (): ((
+  id: LocaleKey, // only accepts valid keys, not any string
+  values?: Record<string, PrimitiveType>
+) => string) => {
+  const intl = useIntl();
+  return (id, values) => intl.formatMessage({ id: id as string }, values);
+};
+
+export const useFormatDate = (): ((
+  value: number | Date,
+  options?: Intl.DateTimeFormatOptions & { format?: string }
+) => string) => {
+  const intl = useIntl();
+  return (value, options) => intl.formatDate(value, options);
+};
+
+// return type on this signature enforces that all languages have the same translations defined
+export const importMessages = (
+  locale: SupportedLocales
+): Promise<LocaleMessages> => {
+  switch (locale) {
+    case 'en':
+      return import(
+        /* webpackMode: "lazy", webpackChunkName: "en_json" */

```

Comment:
perhaps don't need to break these out to separate bundles, but I rather like it. allows the browser to download and cache separately